### PR TITLE
[PM-12607] Move key rotation to km ownership

### PR DESCRIPTION
**Pull request by quexten**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-12607

## 📔 Objective
Move ownership of userkey rotation and legacy key migration to km.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/web/src/app/key-management/key-rotation/user-key-rotation-api.service.ts
Comment (mzieniukbw) [inline apps/web/src/app/key-management/key-rotation/user-key-rotation-api.service.ts:1]: The code ownership file says that key-management for web lives in `apps/web/src/key-management`, not in apps/web/src/**app**/key-management

**Review by mzieniukbw — CHANGES_REQUESTED** [review]
The files moved path does not match the code ownership for KM

File: apps/web/src/app/key-management/key-rotation/user-key-rotation-api.service.ts
Comment (quexten) [reply apps/web/src/app/key-management/key-rotation/user-key-rotation-api.service.ts:1]: Nice catch! Seems like a mistake in the CODEOWNERS file. Updated it to be consistent with all other teams.

**Review by djsmith85 — CHANGES_REQUESTED** [review]

File: libs/auth/src/common/abstractions/index.ts [hunk 1]
```diff
@@ -3,4 +3,4 @@ export * from "./login-email.service";
 export * from "./login-strategy.service";
 export * from "./user-decryption-options.service.abstraction";
 export * from "./auth-request.service.abstraction";
-export * from "./user-key-rotation-data-provider.abstraction";
+export * from "../../../../key-management/src/abstractions/user-key-rotation-data-provider.abstraction";
```
[the remark this answers is not in the document]
Comment (quexten) [reply libs/auth/src/common/abstractions/index.ts]: Fixed, thanks!

**Review by mzieniukbw — APPROVED** [review]

**Review by mzieniukbw — APPROVED** [review]

**Review by rr-bw — APPROVED** [review]

**Review by BTreston — APPROVED** [review]

**Review by ike-kottlowski — APPROVED** [review]

**Pull request merged**